server/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -1014,10 +1014,10 @@ private <T extends QueryVectorBuilder> void registerQueryVectorBuilder(QueryVect
     private void registerFetchSubPhases(List<SearchPlugin> plugins) {
         registerFetchSubPhase(new ExplainPhase());
         registerFetchSubPhase(new StoredFieldsPhase());
+        registerFetchSubPhase(new FetchFieldsPhase());
         registerFetchSubPhase(new FetchDocValuesPhase());
         registerFetchSubPhase(new ScriptFieldsPhase());
         registerFetchSubPhase(new FetchSourcePhase());
-        registerFetchSubPhase(new FetchFieldsPhase());
```

Comment:
is this related to the NPE in #106617 ? I'd like to better understand where we rely on ordering and why that needs changing.

Reply:
I reverted this...